### ISSUE TITLE
GD-182: `assert_vector.is_equal_approx()` gives false negative

### ISSUE THREAD

**Issue by MJacred** [MikeSchulze/gdUnit4#182]
**The used GdUnit4 version:**
4.1.0

**The used Godot version:**
4.0.2

**Describe the bug**
```gdscript
const encoded_rgb8:Vector3 = Vector3(0.0, 0.878431, 0.105882) # wrote false z value to showcase issue
const approx_vec3:Vector3 = Vector3(0.000001, 0.000001, 0.000001)
const height:float = 120.0
const approx_float:float = 0.000001


func test_encode_height_to_rgb8_unorm() -> void:
	var color:Color = Source.encode_height_to_rgb8_unorm(height)
	var actual:Vector3 = Vector3(color.r, color.g, color.b)
	prints("expected:", encoded_rgb8, "; got:", actual)
	assert_vector3(actual).is_equal_approx(encoded_rgb8, approx_vec3)
```

prints
```sh
expected: (0, 0.878431, 0.105882) ; got: (0, 0.878431, 0.505882)
```

![image](https://github.com/MikeSchulze/gdUnit4/assets/6639237/b40dd452-02d9-403e-992c-0ba9f4b542f0)

**Steps to Reproduce**

```gdscript
func test_is_equal_approx_() -> void:
	var actual   := Vector3(0.0, 0.878431, 0.505882)
	var expected := Vector3(0.0, 0.878431, 0.105882)
	var approx   := Vector3(0.000001, 0.000001, 0.000001)
	
	assert_vector(actual).is_equal_approx(expected, approx)
```

**Comment by MikeSchulze** [comment]
very interesting seams to be a Godot bug
```
	var actual   := Vector3(0.000001, 0.878431, 0.505882)
	var to := Vector3(0.000002, 0.878431, 0.105882)
	prints("actual:", actual)
	prints("to    :", to)
	prints(actual <= to, actual.x <= to.x,  actual.y <= to.y, actual.z <= to.z)
```

actual: (0.000001, 0.878431, 0.505882)
to    : (0.000002, 0.878431, 0.105882)
true (true true false)

The operator <= should end with false but is true

From official documentation:
Compares two Vector3 vectors by first checking if the X value of the left vector is less than or equal to the X value of the right vector. If the **X values are exactly equal,** then it repeats this check with the Y values of the two vectors, and then with the Z values. This operator is useful for sorting vectors.

The documentation is wrong x must be equal or less!

**Comment by MikeSchulze** [comment]
I do workaround this and will provide a fix with 4.1.1
